shielder/contract/lib.rs

```diff
@@ -111,6 +112,85 @@ pub mod contract {
                     )
                     .returns::<Result<(), PSP22Error>>()
                     .invoke()?,
+                OpPub::DepositRelayer {
+                    amount,
+                    token,
+                    user,
+                    fee,
+                    fee_token,
+                    relayer,
+                } => {
+                    build_call::<DefaultEnvironment>()
+                        .call(AccountId::from(token.bytes))
+                        .call_v1()
+                        .gas_limit(0)
+                        .transferred_value(0)
+                        .exec_input(
+                            ExecutionInput::new(Selector::new(ink::selector_bytes!(
+                                "PSP22::transfer_from"
+                            )))
+                            .push_arg(AccountId::from(user.bytes))
+                            .push_arg(self.env().account_id())
+                            .push_arg(amount)
+                            .push_arg([].to_vec() as ink::prelude::vec::Vec<u8>),
+                        )
+                        .returns::<Result<(), PSP22Error>>()
+                        .invoke()?;
```

Comment:
please reduce repetitions across this module - you can crate functions dedicated for `transfer` and `transfer_from` that would hide the whole boilerplate